usr/lib/python3/dist-packages/systemd_boot_manager/__init__.py

```diff
@@ -574,7 +574,11 @@ def get_kernel_versions():
         error("NO KERNELS FOUND IN /boot")
         failure(1)
     for each in KERNELS:
-        KERNELS[KERNELS.index(each)] = each.split("-")[-1]
+        if "xanmod" in each:
+            # TODO @batcastle please find a better solution, dude
+            KERNELS[KERNELS.index(each)] = "-".join(each.split("-")[1:])
+        else:
+            KERNELS[KERNELS.index(each)] = each.split("-")[-1]
```

Comment:
Your way of doing this is superior to mine, as it works with Xanmod, Liquorix, and the mainline kernel. The problem I see right now, is it isn't sorting kernel versions correctly. For instance, I just tried this on my computer and 6.8.8 got marked as the latest kernel, not 6.8.10. We need to work around this.

I DID have a much better solution, which worked with Liquorix, Xanmod, and mainline kernels, and kept them in order, but in my infinite wisdom, I lost them in my undo history. So, I will be redoing that, and posting a suggestion of what to do somewhere.

Reply:
:+1: thanks

Reply:
Something like this should work. This code block would replace the entire function in question.
```python3
def get_kernel_versions():
    """Get all kernel versions"""
    KERNELS = os.listdir(BOOT_DIR)
    # Filter down to just the Kernel Images
    for each in range(len(KERNELS) - 1, -1, -1):
        if "vmlinuz-" not in KERNELS[each]:
            del KERNELS[each]
        # Remove dpkg-tmp images
        elif ".dpkg-tmp" in KERNELS[each][-9:]:
            del KERNELS[each]
    # Check to make sure we have kernels to work with
    if len(KERNELS) < 1:
        error("NO KERNELS FOUND IN /boot")
        failure(1)
    kernel_verts = {}
    for each in KERNELS:
        vert = each.split("-")[1:]
        if len(vert) > 1:
            patch_version = 1
            while f"{ vert[0] }.{ patch_version }" in kernel_verts:
                patch_version += 1
            key = f"{ vert[0] }.{ patch_version }"
        else:
            key = vert[0]
        kernel_verts[key] = "-".join(vert)

    # Sort remaining kernels, get latest
    KERNELS = sorted(list(kernel_verts.keys()), key=LooseVersion)
    output = []
    for each in KERNELS:
    	output.append(kernel_verts[each])
    return output
```
Essentially, the difference here is that non-mainline kernels will usually have some sort of suffix on their name (`-xanmod`, `-liquorix`, etc.) We want that split off the main version code. From there, just append a new patch version to the kernel version code.

While those can be sorted easily, anything calling this function won't understand what we mean when we say "6.8.9.1 > 6.8.9". So, instead, we need to correlate these new version codes we create with the version string they originated from. From there, we can create a new list, properly sorted, based off the list with the extra patch versions, and pass that back to our parent code.

I did test this with my machine, installing mainline, Xanmod, and Liquorix kernels, just Xanmod and mainline, and just mainline. Everything works as it should.

Reply:
merged now